apisix/core/response.lua

```diff
@@ -80,7 +115,20 @@ function resp_exit(code, ...)
     end
 
     if idx > 0 then
-        ngx_print(concat_tab(t, "", 1, idx))
+        local response = concat_tab(t, "", 1, idx)
+        local _, err = decode_json(response)
+        if not err then
+            local content_type = ngx_header["Content-type"]
+            local accept_header = ngx.req.get_headers()["Accept"]
```

Comment:
We can use https://github.com/apache/apisix/blob/c0c906eb4d62886ffa4b0938b017dc4810cfe639/apisix/core/request.lua#L104 which gets the header from cache

Reply:
fixed by [get Accept header from cache](https://github.com/apache/apisix/pull/8587/commits/5e439faa537a97d825699059cc71e3b17df78aa9)